File: 3137-minimum-number-of-operations-to-make-word-k-periodic/3137-minimum-number-of-operations-to-make-word-k-periodic.cpp
class Solution {
public:
    int minimumOperationsToMakeKPeriodic(string word, int k) {
        int N = word.size();
        
        map<string, int> count;
        
        for(int i=0; i<=N-k; i+=k) {
            count[word.substr(i, k)]++;
        }
        
        int mx = 0;
        for(auto e : count) {
            mx = max(mx, e.second);
        }
        
        return N/k - mx;
    }
};

File: 3137-minimum-number-of-operations-to-make-word-k-periodic/3137-minimum-number-of-operations-to-make-word-k-periodic.java
class Solution {
    public int minimumOperationsToMakeKPeriodic(String word, int k) {
        HashMap<String,Integer> hm=new HashMap<>();
        int maxcount=0;
        for(int i=0;i<=word.length()-k;i+=k){
            hm.put(word.substring(i,i+k),hm.getOrDefault(word.substring(i,i+k),0)+1);
            maxcount=Math.max(maxcount,hm.get(word.substring(i,i+k)));
        }
        return word.length()/k-maxcount;
    }
}